pkg/virtualgateway/membership_designator.go

```diff
@@ -85,26 +86,65 @@ func (d *membershipDesignator) DesignateForGatewayRoute(ctx context.Context, obj
 	}
 
 	var vgCandidates []*appmesh.VirtualGateway
+
 	for _, vgObj := range vgList.Items {
-		selector, err := metav1.LabelSelectorAsSelector(vgObj.Spec.NamespaceSelector)
-		if err != nil {
+		if matches, err := matchesNamespaceSelector(objNS, &vgObj); err != nil {
```

Comment:
We pick up VGs by namespace of GatewayRoute only if GatewayRouteSelector isn't defined, isn't it? Why do we still need a namespaceSelector? and the order in which we are looking for VGs is not confirming that as well..

Reply:
I see that we're taking union of both. Should be Ok..